maintenance/scripts/channel_minmax_to_table.py

```diff
@@ -0,0 +1,122 @@
+# -----------------------------------------------------------------------------
+#  Copyright (C) 2018 University of Dundee. All rights reserved.
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+# Script uses Channel min/max intensity to generate OMERO.table on Plate
+
+import argparse
+import omero
+from omero.rtypes import rstring
+import omero.grid
+from omero.gateway import BlitzGateway
+
+
+NAMESPACE = "openmicroscopy.org/omero/bulk_annotations"
+
+
+def run(username, password, plate_id, host, port):
+
+    conn = BlitzGateway(username, password, host=host, port=port)
+    try:
+        conn.connect()
+
+        # Create a name for the Original File
+        tablename = "Channels_Min_Mex_Intensity"
+
+        # Go through all wells in Plate, adding row for each
+        plate = conn.getObject("Plate", plate_id)
+        wellIds = []
+        rowData = []
+        chCount = 0
+        for well in plate._listChildren():
+            well = omero.gateway.WellWrapper(conn, well)
+            image = well.getImage()
+            if image is None:
+                continue
+            wellIds.append(well.id)
+            chCount = image.getSizeC()
+            row = []
+            print "well, image", well.id, image.id
+            for ch in image.getChannels():
+                row.append(long(ch.getWindowMin()))
+                row.append(long(ch.getWindowMax()))
+            image._re.close()
```

Comment:
you should load the rendering def and retrieve the window min/max
Starting a rendering engine is costly
you could for example use method from the RenderingSettings Service